selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/theia/TheiaProjectTree.java

```diff
@@ -128,4 +165,47 @@ public void openItem(String itemPath) {
     waitItemSelected(itemPath);
     doubleClickOnItem(itemPath);
   }
+
+  public boolean isItemVisible(String itemPath) {
+    final String itemId = getProjectItemId(itemPath);
+
+    return seleniumWebDriverHelper.isVisible(By.id(itemId));
+  }
+
+  public void expandPath(String expandPath) {
+    List<String> itemsPaths = splitProjectPath(expandPath);
+
+    for (int i = 0; i < itemsPaths.size(); i++) {
+      String currentItemPath = itemsPaths.get(i);
+
+      if (isItemExpanded(currentItemPath)) {
+        continue;
+      }
+
+      waitItem(currentItemPath);
+      openItem(currentItemPath);
+    }
+  }
+
+  public void expandPathAndOpenFile(String expandPath, String fileName) {
+    expandPath(expandPath);
+    String pathToFile = expandPath.replace('.', '/') + '/' + fileName;
+    waitItem(pathToFile);
+    openItem(pathToFile);
+    theiaEditor.waitEditorTab(fileName);
+    theiaEditor.waitTabSelecting(fileName);
+    theiaEditor.waitActiveEditor();
+  }
+
+  private List<String> splitProjectPath(String path) {
```

Comment:
It's not quite clear what this method does. Could you, please, leave an example of transformation in the method comment?